src/py/aspen/covidhub_import/implementation.py

```diff
@@ -0,0 +1,139 @@
+                ConsensusGenome.id.desc(),
+            )
+            .options(
+                joinedload(ConsensusGenome.sample_fastqs)
+                .joinedload(SampleFastqs.czb_id)
+                .joinedload(CZBID.genome_submission_info),
+            )
+        )
+
+        external_accessions = {
+            consensus_genome.sample_fastqs.czb_id.external_accession
+            for consensus_genome in consensus_genomes
+            if isinstance(consensus_genome.sample_fastqs.czb_id, DphCZBID)
+        }
+        external_accessions_to_samples: Mapping[str, Sample] = {
+            sample.private_identifier: sample
+            for sample in session.query(Sample).filter(
+                and_(
+                    Sample.submitting_group == group,
+                    Sample.private_identifier.in_(external_accessions),
+                )
+            )
+        }
+
+        for consensus_genome in consensus_genomes:
+            czbid = consensus_genome.sample_fastqs.czb_id
+            if not isinstance(czbid, DphCZBID):
+                warnings.warn(
+                    "consensus genome has non-dphczbid"
+                    f"{consensus_genome.sample_fastqs.czb_id}"
+                )
+                continue
+            external_accession = czbid.external_accession
+            sample = external_accessions_to_samples.get(external_accession, None)
+            if sample is None:
+                sample = Sample(
+                    submitting_group=group, private_identifier=external_accession
+                )
+                session.add(sample)
+
+            sample.original_submission = {}
+            sample.public_identifier = (
+                f"USA/{czbid.submission_base}/{czbid.collection_date.year}"
+            )
+            sample.sample_collected_by = project.originating_lab
+            sample.sample_collector_contact_address = project.originating_address
+            sample.collection_date = czbid.collection_date
+            sample.location = project.location
+            sample.division = "California"
+            sample.country = "USA"
+            sample.organism = "SARS-CoV-2"
+
+            if sample.uploaded_pathogen_genome is None:
+                sample.uploaded_pathogen_genome = UploadedPathogenGenome(
+                    sample=sample,
+                )
+                session.add(sample.uploaded_pathogen_genome)
+            sample.uploaded_pathogen_genome.num_unambiguous_sites = (
+                consensus_genome.recovered_sites
+            )
+            sample.uploaded_pathogen_genome.num_mixed = consensus_genome.ambiguous_sites
+            sample.uploaded_pathogen_genome.num_missing_alleles = (
+                consensus_genome.missing_sites
+            )
+            sample.uploaded_pathogen_genome.sequencing_depth = (
+                consensus_genome.avg_depth
+            )
+            sample.uploaded_pathogen_genome.sequence = consensus_genome.fasta
```

Comment:
this is where I could really use your help, @jackkamm

Reply:
I made a couple separate comment threads for `collection_date` and `location`. Otherwise it looks good to me.